R/bernoulli.R

```diff
@@ -157,3 +157,53 @@ cdf.bernoulli <- function(d, x, ...) {
 quantile.bernoulli <- function(d, p, ...) {
   qbinom(p = p, size = 1, prob = d$p)
 }
+
+#' Fit a bernoulli distribution to data
+#'
+#' @param d A `bernoulli` object.
+#' @param x A vector of zeroes and ones
+#'
+#' @return a `bernoulli` object
+#' @export
+fit_mle.bernoulli <- function(d, x, ...) {
+  ss <- suff_stat(d, x, ...)
+  bernoulli(p = ss$successes / (ss$successes + ss$failures))
+}
+
+#' Compute the sufficient statistics for a bernoulli distribution from data
+#'
+#' @inherit beroulli
+#'
+#' @export
+suff_stat.bernoulli <- function(d, x, ...) {
+  valid_x <- (x %in% c(0L, 1L))
+  if(any(!valid_x)) stop("`x` contains elements other than 0 or 1")
+  list(successes = sum(x == 1), failures = sum(x == 0))
+}
+
+#' Compute the likelihood of a bernoulli distribution given data
+#'
+#' @inheritParams fit_mle.bernoulli
+#'
+#' @return the likelihood
+#' @export
+
+likelihood.bernoulli <- function(d, x, ...) {
+  prod(dbinom(x = x, size = 1, prob = d$p))
```

Comment:
Two thoughts:

1. Let's use `pdf()` here rather than `dbinom()`.
2. Products of probabilities tend to underflow. I would compute the sum of log probabilities and then exponential it at the end. So this can just call `log_likelihood()`.